src/lib/features/project/project-service.ts

```diff
@@ -1282,21 +1292,18 @@ export default class ProjectService {
     async statusJob(): Promise<void> {
         const projects = await this.projectStore.getAll();
 
-        const statusUpdates = await Promise.all(
-            projects.map((project) => this.getStatusUpdates(project.id)),
-        );
-
-        await Promise.all(
-            statusUpdates.map((statusUpdate) => {
-                return this.projectStatsStore.updateProjectStats(
-                    statusUpdate.projectId,
-                    statusUpdate.updates,
-                );
-            }),
-        );
+        // run one project status update at a time every
+        void batchExecute(projects, 1, 30_000, async (project) => {
```

Comment:
This will relax the load on DB when you have many projects. Previously each project was running ~10 queries. Now we spread it over time

Reply:
Previously status job was waiting for the completion of all status updates but now we early exit without waiting. batchExecute runs with setTimeout over many minutes but we don't wait on the final result upon job time reporting